### PR TITLE
[dataprotection] Support for PostgreSQLFlexibleServer and MySQL and code cleanup

### PR DESCRIPTION
**Pull request by zubairabid**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

* Added dataprotection support for PostgreSQLFlexibleServer and MySQL workloads: new manifests, code cleanup.
* `az dataprotection backup-instance update-msi-permissions`: New parameter `--target-storage-account-id` for Restore, support Restore for new workloads, code cleanup.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dataprotection

**Comment by AllyW** [comment]
> ## ⚠️ Release Suggestions
> ### Module: dataprotection
> * Update version to `1.0.0b2` in setup.py
> * Set `azext.isPreview` to `true` in azext_dataprotection/azext_metadata.json
> * Remove `azext.isExperimental: true` in azext_dataprotection/azext_metadata.json
> 
> #### Notes
> * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

Hi @zubairabid please follow the release suggestions here for your module version config according to cli extension's version schema: https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md

**Comment by zubairabid** [comment]
> > ## ⚠️ Release Suggestions
> > ### Module: dataprotection
> > 
> > * Update version to `1.0.0b2` in setup.py
> > * Set `azext.isPreview` to `true` in azext_dataprotection/azext_metadata.json
> > * Remove `azext.isExperimental: true` in azext_dataprotection/azext_metadata.json
> > 
> > #### Notes
> > 
> > * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> > * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> > * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)
> 
> Hi @zubairabid please follow the release suggestions here for your module version config according to cli extension's version schema: https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md

Hi @AllyW I removed the isExperimental: true. We want this to be a stable release, are there any other changes I need to make.

File: linter_exclusions.yml [hunk 1]
```diff
@@ -540,6 +540,11 @@ dataprotection backup-instance restore initialize-for-data-recovery-as-files:
     target_blob_container_url:
       rule_exclusions:
       - option_length_too_long
+dataprotection backup-instance update-msi-permissions:
+  parameters:
+    target_storage_account_id:
+      rule_exclusions:
+      - option_length_too_long
```
Comment (evelyn-ys) [inline linter_exclusions.yml:547]: Can you add a short option for this parameter instead of adding exclusion rule?

Comment (zubairabid) [reply linter_exclusions.yml:547]: We can't think of any meaningful alternative name that's also user-friendly. Avoiding `target-sa-id` as it isn't clear what sa is and `target-storage-id` as we need to specify storage account.

**Comment by zubairabid** [comment]
> > ## ⚠️ Release Suggestions
> > ### Module: dataprotection
> > 
> > * Update version to `1.0.0` in setup.py
> > * Remove `azext.isExperimental: true` in azext_dataprotection/azext_metadata.json
> > 
> > #### Notes
> > 
> > * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> > * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> > * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)
> 
> @zubairabid new version has been calculated as stable release. Please adjust the code accordingly

@AllyW the current version is already 1.0.0, just the isExperimental tag had not been removed.
image

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**